src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -57,6 +57,66 @@ private static unsafe float MathFMinMagnitude(float x, float y)
             float ax = MathF.Abs(x), ay = MathF.Abs(y);
             return (ax < ay) || float.IsNaN(ax) || (ax == ay && *(int*)&x < 0) ? x : y;
         }
+
+        private static unsafe int SingleToInt32(float f) => *(int*)&f;
+
+        private static unsafe float Int32ToSingle(int i) => *(float*)&i;
+
+        private static float AnotherSingleNaN = Int32ToSingle(-8388607);
+
+        /// <summary>Loads a variety of special values (e.g. NaN) into random positions in <paramref name="x"/>.</summary>
+        private static void SetSpecialValues(Span<float> x)
+        {
+            // NaN
+            x[s_random.Next(x.Length)] = float.NaN;
```

Comment:
There's at least 5 different interesting `NaN` that would be good to cover here...

In particular, the 4 `NaN` kinds and the inclusive range of their bit patterns are:
* `-qNaN` - `[0xFFC0_0000, 0xFFFF_FFFF]`
* `+qNaN` - `[0x7FC0_0000, 0x7FFF_FFFF]`
* `-sNaN` - `[0xFF00_0001, 0xFFBF_FFFF]`
* `+sNaN` - `[0x7F00_0001, 0x7FBF_FFFF]`

Where for historical reasons, the canonical forms of each of these have historically been:
`-qNaN` - `0xFFC0_0000`
`+qNaN - `0x7FC0_0000`
`-sNaN` - `0xFFA0_0000`
`+sNaN - `0x7FA0_0000`

-- This is because for `qNaN`, it's the bit that distinguishes it as quiet and leaves the payload empty. While for `sNaN`, it's using the most significant payload bit, since at least one payload bit needs to be set to distinguish it from infinity.

The 5th explicitly interesting value is `AllBitsSet`, which is `0xFFFF_FFFF` (a `-qNaN`). We could additionally test random payload bit patterns, but they are less interesting.